include/WCSimDetectorConstruction.hh

```diff
@@ -88,18 +88,35 @@ public:
 
   G4double GetPMTSize1() {return WCPMTSize;}
 
-  G4float GetPMTQE(G4float, G4int, G4float, G4float, G4float);
-  G4float GetPMTCollectionEfficiency(G4float theta_angle) { return GetPMTPointer()->GetCollectionEfficiency(theta_angle); };
+  G4float GetPMTQE(G4String,G4float, G4int, G4float, G4float, G4float);
+  G4float GetPMTCollectionEfficiency(G4float theta_angle, G4String CollectionName) { return GetPMTPointer(CollectionName)->GetCollectionEfficiency(theta_angle); };
 
-  WCSimPMTObject *CreatePMTObject(G4String);
-
-  WCSimPMTObject *  PMTptr;
-  void    SetPMTPointer(WCSimPMTObject* PMT) {PMTptr = PMT;} //currently you can only save one PMT here. When we move to multiple PMTs as a future upgrade, this can be changed to an array of PMT pointers.
-  WCSimPMTObject*  GetPMTPointer(){return PMTptr;}
+  WCSimPMTObject *CreatePMTObject(G4String, G4String);
 
+  G4int total_array_number;
+  G4int array_index;
+  int ODPMTCounter;
+  WCSimPMTObject * PMTptr[10];
+  G4String CollectionNameArray[10];
+ 
+ void SetPMTPointer(WCSimPMTObject* PMT, G4String CollectionName) {
```

Comment:
Ok, I have updated the code to use a map (though, not an existing one). The map uses CollectionName as the key and stores/returns the PMTObject pointer.